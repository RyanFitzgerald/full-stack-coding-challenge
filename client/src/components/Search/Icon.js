import styled from 'styled-components';

const Icon = styled.div`
  bottom: 0;
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 50px;

  svg {
    margin-top: 12px;
  }
`;

export default Icon;