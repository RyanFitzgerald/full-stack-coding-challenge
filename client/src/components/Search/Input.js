import styled from 'styled-components';

const Input = styled.input`
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  box-shadow: 0 1px 10px rgba(67, 66, 97, 0.5);
  outline: none;
  padding: 15px 15px 15px 50px;
  width: 100%;

  @media only screen and (max-width: 480px) {
    padding-right: 15px;
  }
`;

export default Input;