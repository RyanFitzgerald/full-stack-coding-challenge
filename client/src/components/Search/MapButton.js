import styled from 'styled-components';

const TypeButton = styled.button`
  background: ${props => props.active ? '#e5e5e5' : '#fafafa'};
  border: 1px solid #e6e6e6;
  border-bottom-right-radius: 4px;
  border-top-right-radius: 4px;
  bottom: 0;
  color: #696d80;
  cursor: pointer;
  padding: 0;
  position: absolute;
  right: 0;
  top: 0;
  transition: 0.25s ease all;
  width: 100px;

  &:hover {
    background: #e5e5e5;
  }

  @media only screen and (max-width: 480px) {
    bottom: -100%;
    height: 40px;
    left: auto;
    right: 0;
    top: auto;
    width: 50%;
  }
`;

export default TypeButton;