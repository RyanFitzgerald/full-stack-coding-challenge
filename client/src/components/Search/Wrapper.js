import styled from 'styled-components';

const Wrapper = styled.div`
  margin-top: 30px;
  position: relative;
`;

export default Wrapper;