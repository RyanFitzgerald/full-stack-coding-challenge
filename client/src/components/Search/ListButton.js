import styled from 'styled-components';
import MapButton from './MapButton';

const TypeButton = styled(MapButton)`
  border-bottom-right-radius: 0;
  border-right: 0;
  border-top-right-radius: 0;
  right: 100px;

  @media only screen and (max-width: 480px) {
    left: 0;
    right: auto;
  }
`;

export default TypeButton;