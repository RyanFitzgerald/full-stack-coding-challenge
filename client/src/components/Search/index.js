import React from 'react';
import PropTypes from 'prop-types';

// Import components
import Wrapper from './Wrapper';
import Icon from './Icon';
import Input from './Input';
import ListButton from './ListButton';
import MapButton from './MapButton';

// Define the component
function Search({ activeType = 'List', query = '', onQueryChange, onTypeChange }) {
  return (
    <Wrapper>
      <Icon>
        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
          <circle cx="11" cy="11" r="8"></circle>
          <line x1="21" y1="21" x2="16.65" y2="16.65"></line>
        </svg>
      </Icon>
      <Input type="text" name="Search" placeholder="Enter Search Term" defaultValue={query} onChange={(input) => onQueryChange(input.target.value)} />
      <ListButton active={activeType === 'List'} onClick={() => onTypeChange('List')}>List View</ListButton>
      <MapButton active={activeType === 'Map'} onClick={() => onTypeChange('Map')}>Map View</MapButton>
    </Wrapper>
  );
}

// Define the props
Search.propTypes = {
  activeType: PropTypes.string.isRequired,
  query: PropTypes.string.isRequired,
  onQueryChange: PropTypes.func.isRequired,
  onTypeChange: PropTypes.func.isRequired
};

export default Search;