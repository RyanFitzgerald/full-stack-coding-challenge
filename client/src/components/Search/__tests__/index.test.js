import React from 'react';

import Search from '../index';
import Input from '../Input';
import ListButton from '../ListButton';
import MapButton from '../MapButton';

describe('<Search />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <Search/>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('input should equal the query', () => {
    const query = 'test';
    const renderedComponent = shallow(
      <Search query={query} />
    );

    expect(renderedComponent.find(Input).props().defaultValue).toBe(query);
  });

  it('list button should be active', () => {
    const renderedComponent = shallow(
      <Search/>
    );

    expect(renderedComponent.find(ListButton).props().active).toBe(true);
  });

  it('map button should be active', () => {
    const renderedComponent = shallow(
      <Search activeType="Map" />
    );

    expect(renderedComponent.find(MapButton).props().active).toBe(true);
  });
});