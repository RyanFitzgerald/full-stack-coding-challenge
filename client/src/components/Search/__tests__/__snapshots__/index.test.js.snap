// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match the snapshot 1`] = `
.c0 {
  margin-top: 30px;
  position: relative;
}

.c1 {
  bottom: 0;
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 50px;
}

.c1 svg {
  margin-top: 12px;
}

.c2 {
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  box-shadow: 0 1px 10px rgba(67,66,97,0.5);
  outline: none;
  padding: 15px 15px 15px 50px;
  width: 100%;
}

.c4 {
  background: #e5e5e5;
  border: 1px solid #e6e6e6;
  border-bottom-right-radius: 4px;
  border-top-right-radius: 4px;
  bottom: 0;
  color: #696d80;
  cursor: pointer;
  padding: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: 0.25s ease all;
  transition: 0.25s ease all;
  width: 100px;
}

.c4:hover {
  background: #e5e5e5;
}

.c5 {
  background: #fafafa;
  border: 1px solid #e6e6e6;
  border-bottom-right-radius: 4px;
  border-top-right-radius: 4px;
  bottom: 0;
  color: #696d80;
  cursor: pointer;
  padding: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: 0.25s ease all;
  transition: 0.25s ease all;
  width: 100px;
}

.c5:hover {
  background: #e5e5e5;
}

.c3 {
  border-bottom-right-radius: 0;
  border-right: 0;
  border-top-right-radius: 0;
  right: 100px;
}

@media only screen and (max-width:480px) {
  .c2 {
    padding-right: 15px;
  }
}

@media only screen and (max-width:480px) {
  .c4 {
    bottom: -100%;
    height: 40px;
    left: auto;
    right: 0;
    top: auto;
    width: 50%;
  }
}

@media only screen and (max-width:480px) {
  .c5 {
    bottom: -100%;
    height: 40px;
    left: auto;
    right: 0;
    top: auto;
    width: 50%;
  }
}

@media only screen and (max-width:480px) {
  .c3 {
    left: 0;
    right: auto;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
    </svg>
  </div>
  <input
    className="c2"
    defaultValue=""
    name="Search"
    onChange={[Function]}
    placeholder="Enter Search Term"
    type="text"
  />
  <button
    className="c3 c4"
    onClick={[Function]}
  >
    List View
  </button>
  <button
    className="c5"
    onClick={[Function]}
  >
    Map View
  </button>
</div>
`;
